octodns/provider/route53.py

```diff
@@ -465,6 +750,75 @@ def _load_records(self, zone_id):
 
         return self._r53_rrsets[zone_id]
 
+    def _data_for_dynamic(self, name, _type, rrsets):
+        pools = defaultdict(lambda: {'values': []})
+        # Data to build our rules will be collected here and "converted" into
+        # their final form below
+        rules = defaultdict(lambda: {'pool': None, 'geos': []})
+        # Base/empty data
+        data = {
+            'dynamic': {
+                'pools': pools,
+                'rules': [],
+            }
+        }
+
+        # For all the rrsets that comprise this dynamic record
+        for rrset in rrsets:
+            name = rrset['Name']
+            if '-pool.' in name:
+                # This is a pool rrset
+                pool_name = name.split('.', 1)[0][9:-5]
+                if pool_name == 'default':
+                    # default becomes the base for the record and its
+                    # value(s) will fill the non-dynamic values
+                    data_for = getattr(self, '_data_for_{}'.format(_type))
+                    data.update(data_for(rrset))
+                elif rrset['Failover'] == 'SECONDARY':
+                    # This is a failover record, we'll ignore PRIMARY, but
+                    # SECONDARY will tell us what the pool's fallback is
+                    fallback_name = rrset['AliasTarget']['DNSName'] \
+                        .split('.', 1)[0][9:-5]
+                    # Don't care about default fallbacks, anything else
+                    # we'll record
+                    if fallback_name != 'default':
+                        pools[pool_name]['fallback'] = fallback_name
+            elif 'GeoLocation' in rrset:
+                # These are rules
+                _id = rrset['SetIdentifier']
+                # We record rule index as the first part of set-id, the 2nd
+                # part just ensures uniqueness across geos and is ignored
+                i = int(_id.split('-', 1)[0])
+                # Parse the pool name out of _octodns-<pool-name>-pool.
```

Comment:
Ah I see the comment here - might just be good to drop the comment in at each usage since it's a bit unclear without it.